src/status_im2/contexts/wallet/home/view.cljs

```diff
@@ -50,33 +50,35 @@
   (let [top          (safe-area/get-top)
         selected-tab (reagent/atom (:id (first tabs-data)))]
     (fn []
-      [rn/view
-       {:style {:margin-top top
-                :flex       1}}
-       [common.top-nav/view]
-       [rn/view {:style style/overview-container}
-        [quo/wallet-overview temp/wallet-overview-state]]
-       [rn/pressable
-        {:on-long-press #(rf/dispatch [:show-bottom-sheet {:content temp/wallet-temporary-navigation}])}
-        [quo/wallet-graph {:time-frame :empty}]]
-       [rn/view {:style style/accounts-container}
-        [rn/flat-list
-         {:style      style/accounts-list
-          :data       account-cards
-          :horizontal true
-          :separator  [rn/view {:style {:width 12}}]
-          :render-fn  quo/account-card}]]
-       [quo/tabs
-        {:style          style/tabs
-         :size           32
-         :default-active @selected-tab
-         :data           tabs-data
-         :on-change      #(reset! selected-tab %)}]
-       (case @selected-tab
-         :assets       [rn/flat-list
-                        {:render-fn               quo/token-value
-                         :data                    temp/tokens
-                         :key                     :assets-list
-                         :content-container-style {:padding-horizontal 8}}]
-         :collectibles [collectibles/view]
-         [activity/view])])))
+      (let [networks-list (rf/sub [:wallet/network-details])]
+        [rn/view
+         {:style {:margin-top top
+                  :flex       1}}
+         [common.top-nav/view]
+         [rn/view {:style style/overview-container}
+          [quo/wallet-overview (temp/wallet-overview-state networks-list)]]
+         [rn/pressable
+          {:on-long-press #(rf/dispatch [:show-bottom-sheet
+                                         {:content temp/wallet-temporary-navigation}])}
+          [quo/wallet-graph {:time-frame :empty}]]
+         [rn/view {:style style/accounts-container}
+          [rn/flat-list
+           {:style      style/accounts-list
+            :data       account-cards
+            :horizontal true
+            :separator  [rn/view {:style {:width 12}}]
+            :render-fn  quo/account-card}]]
+         [quo/tabs
+          {:style          style/tabs
+           :size           32
+           :default-active @selected-tab
+           :data           tabs-data
+           :on-change      #(reset! selected-tab %)}]
```

Comment:
@J-Son89, often times what I'm going to suggest is not necessary.

This usage of `quo/tabs` only needs to be reprocessed by Reagent when `@selected-tab` is different than the previous value, since all the other props *can* be static. I say *can*, because here we're passing an anonymous function that always generates a new instance in memory, so Reagent will always do work when `wallet.home/view` re-renders, but the `tabs` component is not that lightweight.

Ideally, `on-change` would be extracted to a local binding and the component made a form-2 component.

```clojure
(defn view
  []
  (let [top           (safe-area/get-top)
        selected-tab  (reagent/atom (:id (first tabs-data)))
        on-change-tab #(reset! selected-tab %)]
    (fn []
      (let [networks-list (rf/sub [:wallet/network-details])]
        [rn/view
         ...
         [quo/tabs
          {:style          style/tabs
           :size           32
           :default-active @selected-tab
           :data           tabs-data
           :on-change      on-change-tab}]
         ...]))))
```

Reply:
Yes @J-Son89, I've trying to share this more often to see how other devs respond and if it becomes a more common practice, so we all build an intuition. Sometimes this kind of change can make a noticeable difference, so it's hard for reviewers to catch them without the feedback from development that only the dev working on the PR gets.

Reply:
@ilmotta 

When possible, I define the values only one time outside the render function, since I don't want to re-create the exact same function/value over the rerenders, so I think this is better, but I'm not sure if they have a significant impact on the app :thinking:  maybe only for heavy stuff, idk.

> I've trying to share this more often to see how other devs respond and if it becomes a more common practice,

Just a thought about that "more common practice": I think it's ok to share knoweledge with devs about how reagent works, but I wouldn't like to add a code style rule for this case.

Reply:
Yep @ulisesmac, it should not be a rule because otherwise we would optimizing code without necessity. But in some cases I see we write code that's inefficient where it didn't have to be if we made a minor adjustment.

I went into the trouble of refactoring many places in the mobile app to better take advantage of Reagent's prop diff mechanism, just to see what happens, nothing that I committed. I encourage others explore this idea too. Sometimes the differences were noticeable. React developers have to manually use `memo` and take care of using immutable data to take advantage of this capability, but with Reagent it's a given.

Here the `tabs` component is heavy, with a simple extraction outside the renderer Reagent will completely skip it when re-rendering the parent component if the tab is the same. I think we should code with this efficiency in mind, not as rule :)